packaging/Makefile

```diff
@@ -60,48 +75,65 @@ docker.build.enable:
 	[ -f $$HOME/.docker/config.json ] || touch $$HOME/.docker/config.json
 	if [ -s $$HOME/.docker/config.json ]; then jq -r '. += {experimental: "enabled"}' $$HOME/.docker/config.json > $$HOME/.docker/config.json.new; mv $$HOME/.docker/config.json.new $$HOME/.docker/config.json; else echo '{"experimental": "enabled"}' >$$HOME/.docker/config.json; fi
 
-build.linux: $(SOURCES) build.linux.amd64 build.linux.arm64 build.linux.armv7
+build.linux: build.linux.amd64 build.linux.arm64 build.linux.armv7
+build.linux.amd64: $(addprefix build/linux/amd64/,$(BINARIES))
+build.linux.arm64: $(addprefix build/linux/arm64/,$(BINARIES))
+build.linux.armv7: $(addprefix build/linux/arm/v7/,$(BINARIES))
+
+build.darwin: build.darwin.amd64 build.darwin.arm64
+build.darwin.amd64: $(addprefix build/darwin/amd64/,$(BINARIES))
+build.darwin.arm64: $(addprefix build/darwin/arm64/,$(BINARIES))
+
+build.windows: $(addprefix build/windows/amd64/,$(BINARIES))
 
-build.linux.amd64:
+build/linux/amd64/%:
 	GO111MODULE=$(GO111) \
 	GOOS=linux \
 	GOARCH=amd64 \
 	CGO_ENABLED=$(CGO_ENABLED) \
-	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o build/linux/skipper ../cmd/skipper/*.go
+	go build -ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT_HASH)" -o build/linux/amd64/$(notdir $@) ../cmd/$(notdir $@)/*.go
```

Comment:
What `$(notdir $@)` is for? Or rather how did it work before - looks like only skipper binary was built...

Reply:
Correct! Only the skipper binary was built. You can double-check in the latest release tarballs: https://github.com/zalando/skipper/releases/download/v0.13.213/skipper-v0.13.213-linux-amd64.tar.gz

The docker image was built using other steps (further up) that build all binaries.

`notdir` is a handy way to extract the last part of the current Makefile's target, e.g. here it would be the plain binary name.

E.g. for the task `build/linux/amd64/skipper` `notdir` is `skipper` and `dir` would be `build/linux/amd64/` (https://www.gnu.org/software/make/manual/html_node/File-Name-Functions.html)